src/test/java/com/example/demo/service/TblUserLogicImplTest.java

```diff
@@ -0,0 +1,148 @@
+package com.example.demo.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hamcrest.core.IsEqual;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.example.demo.controllers.SearchController;
+import com.example.demo.daos.TblUserDao;
+import com.example.demo.entities.DisplayUser;
+import com.example.demo.entities.SearchingInfo;
+import com.example.demo.entities.TblUser;
+import com.example.demo.logics.impl.TblUserLogicImpl;
+import com.example.demo.utils.Common;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TblUserLogicImplTest {
+	@InjectMocks
+	private TblUserLogicImpl sut;
+	@Mock
+	private TblUserDao tblUserDao;
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Before
+	public void setUp() {
+		List<TblUser> tblUsers = new ArrayList<>();
+		tblUsers.add(new TblUser());
+
+		List<DisplayUser> displayUsers = new ArrayList<>();
+		displayUsers.add(new DisplayUser(1, "Tran Viet Hung", "01", "1989-12-20", "0123456789", "2016-12-25",
+				"2017-12-25", "Ha Noi"));
+		displayUsers.add(new DisplayUser(20, "Tran Đuc Cong", "01", "1995-01-11", "4567891235", "2017-01-17",
+				"2017-01-31", "Ha Noi"));
+		when(tblUserDao.findByUserNameAndUserPassword(anyString(), anyString())).thenReturn(tblUsers);
+		when(tblUserDao.getNumberOfUsers(anyObject())).thenReturn((long) 24);
+		when(tblUserDao.getListUsers(anyObject(), anyInt(), anyInt())).thenReturn(displayUsers);
+	}
+
+	/**
+	 * Truong hop success
+	 */
+	@Test
+	public void LoginByUsernameAndPasswordTest() {
+		List<TblUser> tblUsers = sut.LoginByUsernameAndPassword("admin", "admin");
+		assertEquals(1, tblUsers.size());
+
+	}
+	/**
+	 * Test paging(int param1, int param2, int param3)
```

Comment:
/**
+  * [IN]
+  * currentPage: 10
+  * totalPage: 100
      ................
+  * [OUT]
+   actual = ......